conans/model/conf.py

```diff
@@ -470,10 +470,12 @@ def copy_conaninfo_conf(self):
         # Reading the list of all the configurations selected by the user to use for the package_id
         package_id_confs = self.get("tools.info.package_id:confs", default=[], check_type=list)
         for conf_name in package_id_confs:
-            value = self.get(conf_name)
-            # Pruning any empty values, those should not affect package ID
-            if value:
-                result.define(conf_name, value)
+            matching_confs = [c for c in BUILT_IN_CONFS if re.match(conf_name, c)] or [conf_name]
```

Comment:
A quick question: If we also want to match the ``user.mycomp.*`` confs, then we would need to iterate better the defined values in the current conf, rather than matching against the ``BUILT_IN_CONFS``? I think this would be a more general approach.